core/src/spotify_uri.rs

```diff
@@ -0,0 +1,677 @@
+use crate::{Error, SpotifyId};
+use std::{borrow::Cow, fmt};
+use thiserror::Error;
+
+use librespot_protocol as protocol;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum SpotifyItemType {
+    Album,
+    Artist,
+    Episode,
+    Playlist,
+    Show,
+    Track,
+    Local,
+    Unknown,
+}
```

Comment:
Personally I would like to still remove this enum as it's a bit redundant with `SpotifyUri` itself.

Meaning all forwarded `SpotifyId` functions to create a uri would be dropped. The user can manually wrap the id in a uri if that's the way they want to do it. The parsing for the uri would directly match the string instead of converting it to the enum and `item_type` would return a `&'static str` to still get the string value of the type.

Most test that seem to be duplicated from `SpotifyId` could then also be dropped, testing the same functionally twice doesn't make to much sense.